kafka-gateway/README.md

```diff
@@ -18,9 +18,47 @@ example properties provided via VM options
 -Dspline.kafka.topic=spline-topic
 ```
 
-### Error recovery
-Currently, all received messages are acknowledged. Errors during message processing are logged. 
-The consumer is idempotent so if needed it can be rewind back, and the failed messages can be re-ingested. 
+### Idempotency
+In accord with Kafka best practices this consumer is indepmotent. 
+In practice, it means that one message can be consumed multiple times without risk of duplication. 
+Spline will recognize that the message was already consumed, and it will simply ignore it.
+
+### Scalability
+By default, Spline Consumer is using one Kafka Consumer. When need to scale arise you can run additional Spline Consumer instances.
+Same rules as for any other Kafka consumers apply.
+
+There is also option to run multiple Kafka Consumers inside one application instance. If for some reason you want to scale vertically.
+You can do this by setting `spline.kafka.consumerConcurrency` to number higher than one.
+
+### Reliability
+To achieve maximum reliability Dead Letter Queue should be enabled, then spline will only commit the message offset 
+once the message data were stored in the database, or the message was sent to the Dead Letter Queue.
+
+Without Dead Letter Queue active the messages that fail will be simply logged and the offset will be committed 
+to not halt the consumption.
+
+There is a retry mechanism that will try to consume the message multiple times before it's considered failed.
+For certain exceptions retry is not attempted. For example, we don't except json parsing to succeed on additional retry.
```

Comment:
```suggestion
To achieve maximum reliability _Dead Letter Queue_ should be enabled. That makes Spline to commit the message offset
only when the message data is either successfully stored to the database, or is sent to the _Dead Letter Queue_.

Without having _Dead Letter Queue_ enabled Spline commits the offset anyway (to avoid repeatedly failing messages clogging processing), and the errors are just logged.

There is a retry mechanism that will try to consume the message multiple times before it's considered failed.
For certain exceptions retry is not attempted. For example, messages that fail on JSON parsing won't be retried.
```